### PR TITLE
Fix JSON parsing when retrieving project integration

### PR DESCRIPTION
**Pull request by chefsalim**
Need to parse back to JSON when retrieving project integration.

Signed-off-by: Salim Alam <salam@chef.io>

tenor-124897091

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-api/src/server/handlers.rs [hunk 1]
```diff
@@ -750,13 +750,25 @@ pub fn get_project_integration(req: &mut Request) -> IronResult<Response> {
     request.set_integration(opi);
 
     match route_message::<OriginProjectIntegrationGet, OriginProjectIntegration>(req, &request) {
-        Ok(integration) => Ok(render_json(status::Ok, &integration.get_body())),
+        Ok(integration) => {
+            let v: serde_json::Value = match serde_json::from_str(&integration.get_body()) {
+                Ok(v) => v,
+                Err(e) => {
+                    debug!("Error parsing to JSON. e = {:?}", e);
+                    return Ok(Response::with((status::UnprocessableEntity, "api-gpi:1")));
```
Comment (raskchanky) [inline components/builder-api/src/server/handlers.rs]: To maintain consistency with the rest of our error codes, this should be `api:gpi:1`

File: components/builder-api/src/server/handlers.rs [hunk 2]
```diff
@@ -750,13 +750,25 @@ pub fn get_project_integration(req: &mut Request) -> IronResult<Response> {
...
             match err.get_code() {
                 ErrCode::ENTITY_NOT_FOUND => Ok(Response::with((status::NotFound))),
                 _ => {
-                    error!("get_project_integration:1, err={:?}", err);
-                    Ok(Response::with(status::InternalServerError))
+                    error!(
+                        "Unexpected error retrieving project integration, err={:?}",
+                        err
+                    );
+                    Ok(Response::with((status::InternalServerError, "api-gpi:2")))
```
Comment (raskchanky) [inline components/builder-api/src/server/handlers.rs]: Same here as above

**Comment by raskchanky** [comment]
tenor-85487593

**Comment by chefsalim** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/281828322).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/281828322).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**